### PR TITLE
Add grpc.RoundRobin support to CT personality

### PR DESCRIPTION
**Pull request by daviddrysdale**
With two underlying resolver implementations:
 - a toy version that returns the names encoded in the target
 - an etcd-based version that monitors for the presence of log servers.

For issue #576

### CODE REVIEW

File: integration/log_prep_test.sh [hunk 1]
```diff
@@ -50,37 +52,34 @@ for ((i=0; i < RPC_SERVER_COUNT; i++)); do
...
-  ./trillian_log_server --rpc_endpoint="localhost:${port}" --http_endpoint='' &
+  ./trillian_log_server ${ETCD_OPTS} --rpc_endpoint="localhost:${port}" --http_endpoint='' &
   pid=$!
   RPC_SERVER_PIDS+=(${pid})
   waitForServerStartup ${port}
 
   # Use the first Log server as the Admin server (any would do)
   if [[ $i -eq 0 ]]; then
-    ADMIN_SERVER="localhost:$port"
+    RPC_SERVER_1="localhost:$port"
```
Comment (AlCutter) [inline integration/log_prep_test.sh]: `${port}`

File: util/resolver_test.go [hunk 2]
```diff
@@ -0,0 +1,81 @@
...
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package util
+
+import (
+	"testing"
+
+	"google.golang.org/grpc/naming"
+
```
Comment (AlCutter) [inline util/resolver_test.go]: could ding this blank line?

File: util/resolver.go [hunk 3]
```diff
@@ -0,0 +1,62 @@
...
+	pendingServers []string
+}
+
+// Next returns a set of updates describing changes to available servers;
+// it will return the original set on first invocation, and block forever
+// thereafter.
+func (fb *FixedBackends) Next() ([]*naming.Update, error) {
+	if len(fb.pendingServers) == 0 {
+		// Block until there is an update.  There will never be an update.
+		select {} // block forever
```
Comment (AlCutter) [inline util/resolver.go]: Would it make sense to have this hang on an unexported `chan` in the struct which gets closed by `Close()` and then returns an error?

File: util/resolver_test.go [hunk 4]
```diff
@@ -0,0 +1,81 @@
...
+		fbr := FixedBackendResolver{}
+		got, err := fbr.Resolve(test.target)
+		if gotErr := (err != nil); gotErr != test.wantErr {
+			t.Errorf("FixedBackendResolver.Resolve(%q)=(%v,%v); want (_,err=%v)", test.target, got, err, test.wantErr)
+		}
+		if err != nil {
+			continue
+		}
+		if diff := pretty.Compare(got, test.want); diff != "" {
+			t.Errorf("FixedBackendResolver.Resolve(%q)=%v; want %v, diff %v", test.target, got, test.want, diff)
```
Comment (AlCutter) [inline util/resolver_test.go]: I've found it helpful to have something like `"... diff -got +want: %v"...` logged for ease of figuring out exactly what's actually happened, you may also prefer to flip the args to `Compare` around too (to get the -want +got).

File: integration/log_prep_test.sh (hunk 1 above)
Comment (daviddrysdale) [inline integration/log_prep_test.sh]: Done.

File: util/resolver_test.go (hunk 2 above)
Comment (daviddrysdale) [inline util/resolver_test.go]: Done.

File: util/resolver_test.go (hunk 4 above)
Comment (daviddrysdale) [inline util/resolver_test.go]: Done.

Comment (codingllama) [inline util/resolver_test.go]: Drive-by comment: please don't flip the args. Most (almost all) test diffs are -got +want, flipping will only make it inconsistent / harder to read.

(Nit: please add a line break before the diff.)

Comment (AlCutter) [inline util/resolver_test.go]: It makes me cry inside.

File: util/resolver.go (hunk 3 above)
Comment (daviddrysdale) [inline util/resolver.go]: Good idea, done.  (I've also made `fixedBackends` internal)
